tests/dlt_broker_from_file_spinner_test.py

```diff
@@ -0,0 +1,199 @@
+    def setUp(self) -> None:
+        self.broker = None
+        _, self.dlt_file_name = tempfile.mkstemp(suffix=b".dlt")
+
+    def tearDown(self) -> None:
+        if self.broker:
+            self.broker.stop()
+        if os.path.exists(self.dlt_file_name):
+            os.remove(self.dlt_file_name)
+
+    def test_broker_with_not_existing_dlt_file(self):
+        """
+        Test DLTBroker could work with not existing dlt file
+
+        1. prepare a file name which does not exist
+        2. start dlt broker to dispatch messages from this not-existing file --> no error
+        3. dlt broker could add context successfully.
+           With ignore_filter_set_ack_timeout=False, exception would get thrown if there is any issue
+           with add_context()
+        4. no message could be dispatched from not existing file and throws out Queue.Empty exception
+        5. dlt_time is 0.0, because it could not be reset according to the latest timestamp of messages
+        """
+        # Remove the dlt file
+        os.remove(self.dlt_file_name)
+        # Start broker with non-existing dlt file
+        self.broker = DLTBroker(
+            ip_address=None,
+            filename=self.dlt_file_name,
+            enable_dlt_time=True,
+            enable_filter_set_ack=True,
+            ignore_filter_set_ack_timeout=False,
+        )
+        self.broker.start()
+        # Add context should report warning message
+        queue = Queue(maxsize=0)
+        self.broker.add_context(queue, filters=None)
+        # Not existing dlt file should not throw any exception out
+        for _ in range(5):
+            with pytest.raises(Empty):
+                queue.get_nowait()
+        # dlt_time is not None, even though it is not reset with latest timestamp from messages
+        self.assertEqual(self.broker.dlt_time(), 0.0)
+
+    def test_broker_with_later_created_dlt_file(self):
+        """
+        Simulate a scenario: first dlt file does not exist, then dlt file is created and written with messages.
+
+        1. delete the dlt file
+        2. start broker
+        3. create the dlt file and write 1 sample message
+            Expectation: 1 message could be dispatched from broker
+        """
+        # 1. delete the dlt file
+        os.remove(self.dlt_file_name)
+        # 2. Start broker with non-existing dlt file
+        self.broker = DLTBroker(
+            ip_address=None,
+            filename=self.dlt_file_name,
+            enable_dlt_time=True,
+            enable_filter_set_ack=True,
+            ignore_filter_set_ack_timeout=False,
+        )
+        self.broker.start()
+        # Add context should report warning message
+        queue = Queue(maxsize=0)
+        self.broker.add_context(queue, filters=None)
+        # 3. Write 1 sample message to the dlt file
+        append_stream_to_file(stream_with_params, self.dlt_file_name)
+        # Expectation: 1 message could be dispatched from broker
+        time.sleep(1)
```

Comment:
please don't sleep in test as possible. It will delay the unit test time.

Reply:
sorry, after writing to dlt file, a short sleep is needed, because I need to first ensure that the messages have been properly written there. The totally unittest takes 59.02 seconds

Reply:
try with 0.1 seconds. should be enough for the IO

Reply:
a call to sync or something similar?

Reply:
thanks for suggestion. instead of fsync, i just added "file.flush" after file.write, then it works out. Now the execution time has been reduced to 38.29 seconds